frontend/src/components/Campaign/CampaignTable.tsx

```diff
@@ -1,41 +1,93 @@
 /* eslint-disable react/react-in-jsx-scope */
 import { ApplicantData } from "@/lib/definitions";
-import { Table, TableCaption, TableContainer, Tbody, Td, Th, Thead, Tr } from "@chakra-ui/react"
-import React from "react";
+import request from "@/util/api";
+import {
+  Table,
+  TableCaption,
+  TableContainer,
+  Tbody,
+  Td,
+  Th,
+  Thead,
+  Tr,
+  useToast,
+} from "@chakra-ui/react";
+import React, { useEffect, useState } from "react";
+import { useQuery } from "react-query";
 
 type CampaignTableProps = {
-    campaignName: string;
-    tableData: ApplicantData[];
-}
+  campaignId: string;
+};
 
-export const CampaignTable: React.FC<CampaignTableProps> = ({ campaignName, tableData }) => {
-    const campaignApplicantData = () => {
-        return tableData.map((data: ApplicantData, index: number) => (
-            <Tr key={index + 1}>
-                <Td>{index + 1}</Td> {/* Needs to be replaced with actual rank from LLM */}
-                <Td>{data.contactDetails.applicantName}</Td> {/* Can either make this a link or add a button to more details */}
-                <Td>13/07/2024</Td>
-                <Td>89%</Td>
-            </Tr>
-        ));
+export const CampaignTable: React.FC<CampaignTableProps> = ({ campaignId }) => {
+  const [campaignName, setCampaignName] = useState("");
```

Comment:
i would combine campaignName and campaignData into a single state, no need to have 2 states. u can define the structure of campaignName as a type or interface